src/k2/module/k23si/client/k23si_client.h

```diff
@@ -174,7 +176,20 @@ class K2TxnHandle {
                 checkResponseStatus(status);
 
                 if constexpr (std::is_same<T, dto::SKVRecord>()) {
-                    return ReadResult<SKVRecord>(std::move(status), std::move(k2response.value));
+                    if (!status.is2xxOK()) {
+                        return seastar::make_ready_future<ReadResult<T>>(ReadResult<T>(std::move(status), SKVRecord()));
+                    }
+
+                    return _client->getSchema(request_cname, request_schema->name, k2response.value.schemaVersion)
+                    .then([s=std::move(status), storage=std::move(k2response.value), request_cname] (auto&& schema_ptr) mutable {
```

Comment:
why copy request_cname here? also above on line #174, why capture request_cname by copy? Could we not capture by ref?